src/js/media-gesture-receiver.js

```diff
@@ -16,6 +15,12 @@ template.innerHTML = `
 </style>
 `;
 
+const resizeObserver = new ResizeObserver((entries) => {
+  for (const entry of entries) {
+    entry.target.rect = entry.target.getBoundingClientRect();
+  }
+});
```

Comment:
I saw this a while back 1 resizeobserver vs multiple per element performs a lot better